mantle/cmd/kola/testiso.go

```diff
@@ -679,55 +679,104 @@ func awaitCompletion(ctx context.Context, inst *platform.QemuInstance, outdir st
-					errchan <- fmt.Errorf("Got EOF from completion channel, %s expected", exp)
-				} else {
-					errchan <- errors.Wrapf(err, "reading from completion channel")
+				errchan <- errors.Wrapf(err, "QEMU unexpectedly exited while awaiting completion")
+			}
+			time.Sleep(1 * time.Minute)
+			errchan <- fmt.Errorf("QEMU exited; timed out waiting for completion")
+		}()
+		go func() {
+			r := bufio.NewReader(qchan)
+			for _, exp := range expected {
+				l, err := r.ReadString('\n')
+				if err != nil {
+					if err == io.EOF {
+						// this may be from QEMU getting killed or exiting; wait a bit
+						// to give a chance for .Wait() above to feed the channel with a
+						// better error
+						time.Sleep(1 * time.Second)
+						errchan <- fmt.Errorf("Got EOF from completion channel, %s expected", exp)
+					} else {
+						errchan <- errors.Wrapf(err, "reading from completion channel")
+					}
+					return
 				}
-				return
+				line := strings.TrimSpace(l)
+				if line != exp {
+					errchan <- fmt.Errorf("Unexpected string from completion channel: %s expected: %s", line, exp)
+					return
+				}
+				plog.Debugf("Matched expected message %s", exp)
 			}
-			line := strings.TrimSpace(l)
-			if line != exp {
-				errchan <- fmt.Errorf("Unexpected string from completion channel: %s expected: %s", line, exp)
-				return
+			plog.Debugf("Matched all expected messages")
+			// OK!
+			errchan <- nil
+		}()
+		go func() {
+			//check for error when switching boot order
+			if booterrchan != nil {
+				if err := <-booterrchan; err != nil {
+					errchan <- err
+				}
 			}
-			plog.Debugf("Matched expected message %s", exp)
-		}
-		plog.Debugf("Matched all expected messages")
-		// OK!
-		errchan <- nil
-	}()
-	go func() {
-		//check for error when switching boot order
-		if booterrchan != nil {
-			if err := <-booterrchan; err != nil {
-				errchan <- err
+		}()
+		err := <-errchan
+		elapsed := time.Since(start)
+		if err == nil {
+			// No error so far, check the console and journal files
+			consoleFile := filepath.Join(outdir, "console.txt")
+			journalFile := filepath.Join(outdir, "journal.txt")
+			files := []string{consoleFile, journalFile}
+			for _, file := range files {
+				fileName := filepath.Base(file)
+				// Check if the file exists
+				_, err := os.Stat(file)
+				if os.IsNotExist(err) {
+					fmt.Printf("The file: %v does not exist\n", fileName)
+					continue
+				} else if err != nil {
+					fmt.Println(err)
+					errchan <- err
+					return elapsed, err
+				}
+				// Read the contents of the file
+				fileContent, err := os.ReadFile(file)
+				if err != nil {
+					fmt.Println(err)
+					errchan <- err
+					return elapsed, err
+				}
+				// Check for badness with CheckConsole
+				warnOnly, badlines := kola.CheckConsole([]byte(fileContent), nil)
+				if len(badlines) == 0 {
+					fmt.Printf("No badness detected in the file: %v\n", fileName)
+				} else {
+					for _, badline := range badlines {
```

Comment:
we should check warnonly in this loop and then print out a different log message accordingly.. i.e. warning or error prefixed message (plog.Warningf or plog.Errorf)

we should not set `err` in this loop I don't think. We can set it at the top of this block in the else with something generic.

```
if !warnOnly {
    err = fmt.Errorf("checkConsole found bad lines in {fileName}")
}
````

Reply:
> We can set it at the top of this block in the else with something generic

or maybe better at the bottom of the block and then just break out of the outer loop so the journal won't get checked if we already found bad lines in the console.

Reply:
I think I corrected it as requested in the [latest push](https://github.com/coreos/coreos-assembler/commit/4d50394dd433f673254938162ac1a535d3026211#diff-34b42d984411d80e1feb78dd983ad0263d34d73464ba6eff2751b0c19c157e9bR764-R769)